plugins/utils/flash.nix

```diff
@@ -6,536 +6,669 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.flash;
-in
-{
-  options.plugins.flash =
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "flash";
+  originalName = "flash.nvim";
+  defaultPackage = pkgs.vimPlugins.flash-nvim;
+
+  maintainers = with maintainers; [
+    traxys
+    MattSturgeon
+  ];
+
+  # TODO: Added 2024-06-17; remove 2024-09-17
+  optionsRenamedToSettings =
     let
-      configOpts = {
-        labels = helpers.defaultNullOpts.mkStr "asdfghjklqwertyuiopzxcvbnm" ''
-          Labels appear next to the matches, allowing you to quickly jump to any location. Labels are
-          guaranteed not to exist as a continuation of the search pattern.
-        '';
-
+      # Combine base option paths with nested mode option paths
+      merged = flatten [
+        opts
+        (mapAttrsToList (mode: extras: map (opt: "${mode}.${opt}") (opts ++ extras)) modes)
+      ];
```

Comment:
Why do you have those variables "out of order" ? I would personally prefer an "imperative-friendly" way.

Reply:
Could you clarify what you mean? Do you mean having `merged` declared before `opts`?

I did that to make it obvious up front what the goal of the let block is, rather than having rather large attr sets/lists being the first thing you see.

Happy to change and/or remedy with comments.

Reply:
Yes, I meant
```
a = ...
b = f(a)
c = f(b)
d = f(a, c)
...
```

I find it more readable. Otherwise you are wondering what these `opts` are. But this is personal preference of course.

Reply:
> I find it more readable. Otherwise you are wondering what these `opts` are. But this is personal preference of course.

Generally, I agree. It was only the size of the other variable definitions that changed my mind this time.

I think because the `opts` is _right there_ on the next line it is ok (this time)?

Reply:
Yes sure, either way is fine. We should stop nit picking (I am looking at myself).

Reply:
> Yes sure, either way is fine. We should stop nit picking (I am looking at myself).

Nothing wrong with being thorough and discussing the best approach in nuanced cases.

Maybe avoid nit-picking new contributes if it is likely to put them off contributing, but that doesn't mean let sloppy code in :smiley: